source/Ship.cpp

```diff
@@ -2376,13 +2393,19 @@ shared_ptr<Ship> Ship::Board(bool autoPlunder)
 		SetTargetShip(shared_ptr<Ship>());
 		bool helped = victim->isDisabled;
 		victim->hull = min(max(victim->hull, victim->MinimumHull() * 1.5), victim->attributes.Get("hull"));
-		victim->isDisabled = false;
-		// Transfer some fuel if needed.
-		if(!victim->JumpsRemaining() && CanRefuel(*victim))
+		if(!victim.get()->IsOutOfEnergy())
+			victim->isDisabled = false;
+		// Transfer some fuel if needed.  If your escort is a fighter it should refuel the fleet to maximum.
+		if(!victim->JumpsRemaining() || (IsYours() && CanRefuel(*victim)))
 		{
 			helped = true;
-			TransferFuel(victim->JumpFuelMissing(), victim.get());
+			if(victim->JumpFuelMissing())
+				TransferFuel(victim->JumpFuelMissing(), victim.get());
+			else
+				TransferFuel(fuel, victim.get());
```

Comment:
@quyykk regarding your comments here -> https://github.com/endless-sky/endless-sky/pull/6726/files#r872256676

That code in `JumpFuelMissing()` is important for this section of code.

This enables fighters to have 1 fuel pod and distribute the fuel.  Without carried ships always showing `0` jump fuel you need to have a minimum of 2 fuel pods.  Mainly because the fighter will not distribute "1 jump of fuel" to the fleet because it assumes you need it for the jump (even if your fighter _doesn't_ have a hyperdrive).